src/applications/bmqstoragetool/m_bmqstoragetool_testutils.h

```diff
@@ -79,206 +79,208 @@ namespace m_bmqstoragetool {
-    /// DeleteRecord records even records have the same GUID as MessageRecord,
-    /// odd ones - not the same. Store list of created records in the specified
-    /// `records`. If `expectOutstandingResult` is `true`, store GUIDs of
-    /// outstanding messages in the specified `expectedGUIDs`. Otherwise store
-    /// GUIDs of confirmed messages.
-    void addJournalRecordsWithOutstandingAndConfirmedMessages(
-        RecordsListType* records,
-        GuidVectorType*  expectedGUIDs,
-        bool             expectOutstandingResult);
-
-    /// Generate sequence of MessageRecord, ConfirmRecord and DeleteRecord
-    /// records. MessageRecord and ConfirmRecord records have the same GUID.
-    /// DeleteRecord records even records have the same GUID as MessageRecord,
-    /// odd ones - not the same. Store list of created records in the specified
-    /// `records`. Store GUIDs of partially confirmed messages in the specified
-    /// `expectedGUIDs`.
-    void addJournalRecordsWithPartiallyConfirmedMessages(
-        RecordsListType* records,
-        GuidVectorType*  expectedGUIDs);
-
-    /// Generate sequence of MessageRecord, ConfirmRecord and DeleteRecord
-    /// records. MessageRecord ConfirmRecord and DeletionRecord records have
-    /// the same GUID. queueKey1 is used for even records, queueKey2 for odd
-    /// ones. Store list of created records in the specified `records`. Store
-    /// GUIDs for `queueKey1` in the specified `expectedGUIDs` if
-    /// `captureAllGUIDs` is `false` or all GUIds otherwise.
-    void addJournalRecordsWithTwoQueueKeys(RecordsListType* records,
-                                           GuidVectorType*  expectedGUIDs,
-                                           const char*      queueKey1,
-                                           const char*      queueKey2,
-                                           bool captureAllGUIDs = false);
-
-    /// Generate sequence of MessageRecord and DeleteRecord
-    /// records with size `numMessages`. Change GUIDs order for 2nd and 3rd
-    /// DeleteRecord. Set message offsets from the given `messageOffsets`.
-    /// Store list of created records in the specified `records`. Store
-    /// confirmed GUIDs in the specified `expectedGUIDs`.
-    void addJournalRecordsWithConfirmedMessagesWithDifferentOrder(
-        RecordsListType*           records,
-        GuidVectorType*            expectedGUIDs,
-        size_t                     numMessages,
-        bsl::vector<unsigned int>& messageOffsets);
-};
-
-/// Output the specified `messageGUID` as a string to the specified `ostream`.
-void outputGuidString(bsl::ostream&            ostream,
-                      const bmqt::MessageGUID& messageGUID,
-                      const bool               addNewLine = true);
-
-/// Value semantic type representing data message parameters.
-struct DataMessage {
-    int         d_line;
-    const char* d_appData_p;
-    const char* d_options_p;
-};
+struct TestUtils {
```

Comment:
For the purpose of code organization I think we should avoid putting all these classes in a struct. These helper Util structs should exist to support classes. I think we should just split `JournalFile`, `DataMessage`, and `FileManagerMock` to their own components in the bmqstoragetool package.

If we want them to be clearly marked for testing purposes only, then I think we should explain that in the component level documentation.

Reply:
Extracted `JournalFile` and `FileManagerMock` and removed `m_bmqstoragetool_testutils`.
`DataMessage` struct and 2 smaller free functions from `TestUtils` could be local for one unit test, so I moved it there.